core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -83,6 +101,34 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
 
   def getFile(blockId: BlockId): File = getFile(blockId.name)
 
+  /**
+   * This should be in sync with
+   * @see [[org.apache.spark.network.shuffle.RemoteBlockPushResolver#getFile(java.lang.String,
+   *   java.lang.String)]]
+   */
+  def getMergedShuffleFile(blockId: BlockId): File = {
```

Comment:
@zhouyejoe This is outdated code. We need to pass the `dirs: Option[Array[String]]`to this method to support discovering location of merged block directories when dynamic resource allocation is enabled.